chore: add chosing an exponentAtPriceOne docs (#4919)
osmosis-labs/osmosis: x/concentrated-liquidity/README.md

$$\Delta x = \Delta \frac {1}{\sqrt P}  L$$

> # Ticks

Author: Changed here https://github.com/osmosis-labs/osmosis/pull/4919/commits/6c443d53540cb2c4bc90fedb4e0a0d5b3d388fd4

One draw back of this implementation is the requirement to create many ticks that will likely never be used. For example, in order to create ticks at 10 cent increments for spot prices greater than _$10000_, a $exponentAtPriceOne$ value of -5 must be set, requiring us to traverse ticks 1-3600000 before reaching _$10,000_. This should simply be an inconvenience and should not present any valid DOS vector for the chain.

> # Chosing an Exponent At Price One Value

Reviewer: Could you please elaborate how does base quote translate to the choice of token0 vs token1? That is, if I want token 0 to be SHIB, wich one of the examples should I use?

Author: Added which asset was asset0 and asset1 here https://github.com/osmosis-labs/osmosis/pull/4919/commits/8556ae63f5c692f8daa0efbc57f358453c3d1202

The creator of a pool is required to choose an exponenetAtPriceOne as one of the input parameters. As explained previously, this value determines how much the spot price increases or decreases when traversing ticks. The following equation will assist in selecting this value:

$$exponentAtPriceOne=log_{10}(\frac{D}{P})$$

$$P=(\frac{baseAssetInUSD}{quoteAssetInUSD})$$

> $$D=P-(\frac{baseAssetInUSD}{quoteAssetInUSD+desiredIncrementOfQuoteInUSD})$$

Author: I think it would be really cool that if a user provides a desired price increment that the exponent gets selected automatically! Otherwise if the user provides an exponent then it is assumed that they know what they are doing. I will create an issue for this

> $$D=P-(\frac{baseAssetInUSD}{quoteAssetInUSD+desiredIncrementOfQuoteInUSD})$$

Author: https://github.com/osmosis-labs/osmosis/issues/4922

## Example 1

SHIB is trading at $0.00001070 per SHIB
BTC is trading at $28,000 per BTC

We want to create a SHIB/BTC concentrated liquidity pool where SHIB is the baseAsset and BTC is the quoteAsset. In terms of the quoteAsset, we want to increment in 10 cent values.

$$P=(\frac{0.00001070}{28,000})=0.000000000382142857$$

$$D=(0.000000000382142857)-(\frac{0.00001070}{28,000+0.10})=0.0000000000000013647910441136$$

$$exponentAtPriceOne=log_{10}(\frac{0.0000000000000013647910441136}{0.000000000382142857})=-5.447159582$$

> We can therefore conclude that we can use a spot price of -5 (slightly under precise) or -6 (slightly over precise) for this base/quote pair and desired price granularity.

Reviewer: ```suggestion
We can therefore conclude that we can use an exponent at price  oneof -5 (slightly under precise) or -6 (slightly over precise) for this base/quote pair and desired price granularity.
```

## Example 2

> Flipping the quoteAsset/baseAsset, lets determine what the exponentAtPriceOne should be. For SHIB, centralized exchanges list prices at the 10^-8, so we will set our desired increment to this value.

Reviewer: ```suggestion
Flipping the quoteAsset/baseAsset, for BTC/SHIB, lets determine what the exponentAtPriceOne should be. For SHIB as a quote, centralized exchanges list prices at the 10^-8, so we will set our desired increment to this value.
```

$$P=(\frac{28,000}{0.00001070})=2616822429$$

$$D=(2616822429)-(\frac{28,000}{0.00001070+0.00000001})=2443345$$

$$exponentAtPriceOne=-log_{10}(\frac{2443345}{2616822429})=-3.0297894598783$$

> We can therefore conclude we can use a spot price of -3 for this base/quote pair and desired price granularity.

Reviewer: ```suggestion
We can therefore conclude that we can use an exponent at price one of -3 for this base/quote pair and desired price granularity.
```

### Example 1

SHIB is trading at $0.00001070 per SHIB
BTC is trading at $28,000 per BTC

We want to create a SHIB/BTC concentrated liquidity pool where SHIB is the baseAsset (asset0) and BTC is the quoteAsset (asset1). In terms of the quoteAsset, we want to increment in 10 cent values.

$$P=(\frac{0.00001070}{28,000})=0.000000000382142857$$

$$D=(0.000000000382142857)-(\frac{0.00001070}{28,000+0.10})=0.0000000000000013647910441136$$

$$exponentAtPriceOne=log_{10}(\frac{0.0000000000000013647910441136}{0.000000000382142857})=-5.447159582$$

We can therefore conclude that we can use an exponent at price one of -5 (slightly under precise) or -6 (slightly over precise) for this base/quote pair and desired price granularity.

### Example 2

Flipping the quoteAsset/baseAsset, for BTC/SHIB, lets determine what the exponentAtPriceOne should be. For SHIB as a quote, centralized exchanges list prices at the 10^-8, so we will set our desired increment to this value.

$$P=(\frac{28,000}{0.00001070})=2616822429$$

$$D=(2616822429)-(\frac{28,000}{0.00001070+0.00000001})=2443345$$

$$exponentAtPriceOne=-log_{10}(\frac{2443345}{2616822429})=-3.0297894598783$$

> We can therefore conclude that we can use an exponent at price one of -3 for this base/quote pair and desired price granularity.

Reviewer: I guess we need to update since exponent at price one is -6 now with 100 tick spacing?
